src/lsp/ext.rs

```diff
@@ -51,6 +53,28 @@ impl Notification for CorelibVersionMismatch {
     const METHOD: &'static str = "cairo/corelib-version-mismatch";
 }
 
+/// Collects versions of LS and it's dependencies.
+#[derive(Debug)]
+pub struct LSVersions;
+
+#[derive(Serialize, Deserialize)]
+pub struct BinWithVersion {
+    pub version: String,
+    pub path: PathBuf,
+}
```

Comment:
```suggestion
#[derive(Serialize, Deserialize)]
pub struct PathAndVersion {
    pub path: PathBuf,
    pub version: String,
}
```

wdyt?

1. Name follows definition, which actually makes sense given this is just a typed named tuple.
2. I feel `with` is not something an English-native would say here.